DepositUI/Pages/Deposit.razor.cs

```diff
@@ -0,0 +1,161 @@
+﻿using DepositUI.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+
+namespace DepositUI.Pages
+{
+    public partial class Deposit
+    {
+        private string mode = "Main";
+
+        private List<DepositCalc> depositDetails;
+        private List<RequestError> requestErrors;
+
+        private string AmountStr { get { return amountStr; } set { amountStr = NumberInputCheck(value); } }
+        private string amountStr;
+
+        private string TermStr { get { return termStr; } set { termStr = NumberInputCheck(value); } }
+        private string termStr;
+
+        private string PercentStr { get { return percentStr; } set { percentStr = NumberInputCheck(value); } }
+        private string percentStr;
+
+        private bool getDepositError;
+        private HttpResponseMessage response = null;
+
+        private List<DepositModel> deposits;
+        private int nextdeposit = 0;
+
+        private int depositId;
+
+        private void Main()
+        {
+            depositDetails = null;
+            mode = "Main";
+        }
+
+        private async Task History()
+        {
+            deposits = await GetDepositModels(0, 16);
+            nextdeposit += deposits.Count;
+            mode = "History";
+        }
+
+        private async Task Details(int depositId)
+        {
+            var model = new GetDepositDetails
+            {
+                DepositId = depositId
+            };
+            var request = new HttpRequestMessage(HttpMethod.Get, "https://localhost:44320/api/depositcalc");
+            request.Content = JsonContent.Create<GetDepositDetails>(model);
+
+            var client = ClientFactory.CreateClient();
+            response = await client.SendAsync(request);
+
+            if (response.IsSuccessStatusCode)
+            {
+                depositDetails = await response.Content.ReadFromJsonAsync<List<DepositCalc>>();
+            }
+            this.depositId = depositId;
+            mode = "Details";
+        }
+
+        private async Task LoadMore()
+        {
+            var loaded = await GetDepositModels(nextdeposit, 16);
+            deposits.AddRange(loaded);
+            nextdeposit += loaded.Count;
+        }
+
+        private async Task GetDepositCalc()
+        {
+            requestErrors = new List<RequestError>();
+            double amount;
+            double persent;
+            int term;
+
+            if (!Double.TryParse(NumberInputCheck(amountStr), out amount))
```

Comment:
Divide different logic to separated methods. For example this method has following logical blocks: 1) validation 2) sending request.